src/BuildScriptGenerator/DotNetCore/ProjectFileProviders/ProbeAndFindProjectFileProvider.cs

```diff
@@ -63,26 +69,82 @@ public string GetRelativePathToProjectFile(RepositoryContext context)
             var allProjects = new List<string>();
+            bool functionProjectExists = false;
+            bool webAppProjectExists = false;
+            bool blazorWasmProjectExists = false;
+
             foreach (var file in projectFiles)
             {
                 allProjects.Add(file);
-                if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                if (ProjectFileHelpers.IsAzureBlazorWebAssemblyProject(sourceRepo, file))
                 {
-                    webAppProjects.Add(file);
+                    azureBlazorWasmProjects.Add(file);
+                    blazorWasmProjectExists = true;
                 }
                 else if (ProjectFileHelpers.IsAzureFunctionsProject(sourceRepo, file))
                 {
                     azureFunctionsProjects.Add(file);
+                    functionProjectExists = true;
+                }
+                else if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                {
+                    webAppProjects.Add(file);
+                    webAppProjectExists = true;
                 }
             }
 
-            projectFile = GetProject(webAppProjects);
-            if (projectFile == null)
+            // Assumption: some build option "--apptype" will be passed to oryx
+            // which will indicate if the app is an azure function type or static type app
+
+            // If there are multiple projects, we will look for --appty to detect corresponding project.
+            // for example azurefunction and blazor both projects can reside
+            // at the same repo, so more than 2 csprojs will be found. Now we will
+            // look for --apptype value to determine which project needs to be built
+            string oryxAppTypeEnv = string.Empty;
+
+            if (_options != null
+                && _options.Value != null
+                && _options.Value.OryxAppType != null)
             {
-                projectFile = GetProject(azureFunctionsProjects);
+                oryxAppTypeEnv = _options.Value.OryxAppType;
+                _logger.LogInformation($"ProbeAndProjectFileProvider: {Constants.OryxAppType} is set to {oryxAppTypeEnv}");
+
+                if (functionProjectExists && oryxAppTypeEnv.ToLower().Contains("functions"))
+                {
+                    projectFile = GetProject(azureFunctionsProjects);
+                }
+                else if (blazorWasmProjectExists
+                    && (oryxAppTypeEnv.ToLower().Contains("static-sites")
+                    || oryxAppTypeEnv.ToLower().Contains("blazor-wasm")))
+                {
+                    projectFile = GetProject(azureBlazorWasmProjects);
+                }
+                else
+                {
+                    _logger.LogDebug($"Invalid value {oryxAppTypeEnv} for env:{Constants.OryxAppType}. Currently, supported values are 'functions', 'blazor-wasm', 'static-sites'");
```

Comment:
This is too late for this kind of validation to happen. You need to do this in `BuildCommnd.IsValidInput`

Reply:
yeah I know .. I was testing something and kept it here ..I will remove it from here